### ISSUE TITLE
member.php

### ISSUE THREAD

**Issue by FaizcommandoAcademy** [FaizcommandoAcademy/AcademicReveiws#1]
<?php
session_start();
require_once __DIR__ . '../assets/src/Facebook/autoload.php';

$fb = new Facebook\Facebook([
  'app_id' => '166436200',
  'app_secret' => '1cf61385fbf2fc580094b9da',
  'default_graph_version' => 'v2.8',
  ]);

$helper = $fb->getRedirectLoginHelper();

$permissions = ['email']; // optional

$loginUrl=null;
  
try {
  if (isset($_SESSION['facebook_access_token'])) {
    $accessToken = $_SESSION['facebook_access_token'];
  } else {
      $accessToken = $helper->getAccessToken();
  }
} catch(Facebook\Exceptions\FacebookResponseException $e) {
  // When Graph returns an error
  echo 'Graph returned an error: ' . $e->getMessage();

    exit;
} catch(Facebook\Exceptions\FacebookSDKException $e) {
   //When validation fails or other local issues
  echo 'Facebook SDK returned an error: ' . $e->getMessage();
    exit;
 }

//to logout from facebook
 if(isset($_GET['stu_status']) && $_GET['stu_status'] === 'logout'){

$url = 'https://www.facebook.com/logout.php?next=' . 'http://http://www.academicreviewsfyp.com//member.php' .
  '&access_token='.$accessToken;
session_destroy();
header('Location: '.$url);

}

if (isset($accessToken)) {

  $_SESSION['user_status']="active";
  $logoutUrl = $helper->getLogoutUrl($session,'http://http://www.academicreviewsfyp.com//member.php');

  if (isset($_SESSION['facebook_access_token'])) {
    $fb->setDefaultAccessToken($_SESSION['facebook_access_token']);
  } else {
    // getting short-lived access token
    $_SESSION['facebook_access_token'] = (string) $accessToken;

      // OAuth 2.0 client handler
    $oAuth2Client = $fb->getOAuth2Client();

    // Exchanges a short-lived access token for a long-lived one
    $longLivedAccessToken = $oAuth2Client->getLongLivedAccessToken($_SESSION['facebook_access_token']);

    $_SESSION['facebook_access_token'] = (string) $longLivedAccessToken;

     //setting default access token to be used in script
    $fb->setDefaultAccessToken($_SESSION['facebook_access_token']);
  }

  // redirect the user back to the same page if it has "code" GET variable
  if (isset($_GET['code'])) {
      header('location:index.php');
  }

  // getting basic info about user
  try {
    $profile_request = $fb->get('/me?fields=name,first_name,last_name,email');
    //permission(FB app live) hone k bad sirf ye field extra add karenge work 
    $profile = $profile_request->getGraphNode()->asArray();
    //geting user_email for many purposes
    $_SESSION['user_email']=$profile['email'];
  } catch(Facebook\Exceptions\FacebookResponseException $e) {
    // When Graph returns an error
    echo 'Graph returned an error: ' . $e->getMessage();
    session_destroy();
    // redirecting user back to app login page
    header("Location: ./");
    exit;
  } catch(Facebook\Exceptions\FacebookSDKException $e) {
    // When validation fails or other local issues
    echo 'Facebook SDK returned an error: ' . $e->getMessage();
    exit;
  }
  
  // printing $profile array on the screen which holds the basic info about user
  print_r($profile);

    // Now you can redirect to another page and use the access token from $_SESSION['facebook_access_token']
} else {
    $_SESSION['user_status']="deactive";
  // replace your website URL same as added in the developers.facebook.com/apps e.g. if you used http instead of https and you used non-www version or www version of your website then you must add the same here
  $loginUrl = $helper->getLoginUrl('http://localhost:8080/AR/member.php', $permissions);
  echo '<a href="' . $loginUrl . '">Log in with Facebook!</a>';
}

?>

<!DOCTYPE html>

<html class="no-js">

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>

<meta http-equiv="content-type" content="text/html;charset=utf-8" />

<?php include('imp_files/head_file.php') ?>

<body class="account_page">

 

<?php include('imp_files/mobileNav.php') ?>

<div id="container">
  <div class="overlay"></div>
  
  <?php include('imp_files/Realmobileheader.php')  ?>
  
  
  <section class="top-header">
    <div id="ad1x1" data-ad-sizes="1x1" data-ad-keyvalues="" data-ad-reload-interval="-1" style=""></div>
    <div id="ad1x2" data-ad-sizes="1x2" data-ad-keyvalues="" data-ad-reload-interval="-1" style=""></div>
    <div id="ad3x3" data-ad-sizes="3x3" data-ad-keyvalues="" data-ad-reload-interval="-1" style=""></div>
    <div id="ad6x6" data-ad-sizes="6x6" data-ad-keyvalues="" data-ad-reload-interval="-1" style=""></div>
    <article id="mtv_lead_1" class="ad leaderboard_ad">
      <div id="ad728" style="text-align: center;"></div>
    </article>
  </section>
  
  

  
<?php include('imp_files/header.php') ?>
  
   
  

  <?php include('imp_files/Dekstop_leftNav.php') ?>
  
  
  
  
  <div id="body" class="slide">
    <div id="mainContent" class="height-col">
      <div class="page-wrap">
        <div class="header">Log In to Academic Review</div>
        <div class="explain"> <b>Go ahead, log in, you’re still anonymous.</b> You’re always anonymous here, but logging in will allow you to rate professors and universities. You can thank us later. </div>
<?php

if (!isset($accessToken)) {
  echo '<a class="fblogin-btn" href="' . $loginUrl . '"><img src="images/fblogin.png"></a>';
}else{  echo '<a class="fblogin-btn" href="?stu_status=logout"><img src="images/fblogout.png"></a>'; }

?>
      </div>
    </div>
  </div>
   
  
 
    <?php include('imp_files/footer.php') ?>
  

<div class = "promotion"> <a class = "open" onclick="javascript:mtvn.btg.Controller.sendLinkEvent({ linkName:'PROMOS:Maximize', linkType:'o' } );">&#9664;</a> <a class = "close" onclick="javascript:mtvn.btg.Controller.sendLinkEvent({ linkName:'PROMOS:Minimize', linkType:'o' } );">×</a>
  <div class = "content"> <img class = "promo-image" src = "../blog.ratemyprofessors.com/wp-content/uploads/2015/01/WNOs6.5_RMP_72x72.jpg">
    <div class = "text-content">
      <h3 class = "title">Wild 'N Out Returns</h3>
      <span class = "description">Wednesday Jan 7 @ 11pm only on MTV2!</span> </div>
    <div class = "clearfix"></div>
    <span id="promoid" style="display:none">0</span> <a class = "more-info" onclick="promoClickEvent();"> get more info about the premiere </a> </div>
</div>

    <?php include('imp_files/liveSearch_func.php') ?>

 
 

 
 

 
 
<script type="text/template" id="loginTemplate">

  <div id="loginMenu" class="lform">
    <div class="header">
      Login to Rate My Professors
    </div>
    <div class="why-signup">
      <span>Go ahead, log in, you're still anonymous.</span> You're always anonymous here, but logging in will save your university and recent searches, helping you find professors faster. You can thank us later.
    </div>
    <form id="loginForm" action="/userauth" method="post">
      <div class="form-element">
        <div class="error email password">Incorrect email or password</div>
        <div class="label">email</div>
        <input type="text" name="pemail" class="pemail"/>
      </div>

      <div class="form-element">
        <div class="label">password</div>
        <input type="text" name="ppassword" class="ppassword" />
      </div>

      <div class="form-element">
        <input type="checkbox" name="remember" value="0"/>
        <label class="rememberme">Remember Me</label>
        <a href="#" class="forgotpw" id="forgotPassword">Forgot my password </a>
      </div>

      <div class="form-element">
        <input type="submit" id="userLogin" value="Login"/>
        <div class="or-block">
          <hr/>
          <div class="ortext">or</div>
        </div>
        <a href="#" id="signupLink">Sign up for an account</a>

      </div>

    </form>

  </div>

</script> 

 
 

 
 
<script type="text/template" id="resetPasswordForm">

  <div id="resetPassword" class="lform">
    <div class="header">Forgot Password</div>
    <p>Enter your login email to reset your password.</p>
    <form action="" method="post" id="resetPasswordField">
      <div class="form-element email-form">
        <div class="label">email</div>
        <input type="text" name="reset_password" id="reset_password">
      </div>
      <div class="form-element">
        <input type="submit" value="Reset password" id="reset_password_submit">
      </div>
    </form>
  </div>

</script> 

 
 

 

 

 
<script type="text/template" id="autocomplete-profitem">
  <li data-id="{{pk_id}}">
    <span class="wrap">
      <span class="main">{{{teacherfullname_s}}}</span>
      <span class="sub">{{schoolname_s}}</span>
    </span>
  </li>
</script> 
<script type="text/template" id="autocomplete-schoolitem">
  <li data-id="{{pk_id}}">
    <span class="wrap">
      <span class="main">{{schoolname_s}}</span>
      <span class="sub">{{schoolcity_s}}, {{schoolstate_s}}  <span class="typeahead-country">{{schoolcountry_s}}</span></span>
    </span>
  </li>
</script> 
 

<script type="text/javascript">
      window.RMP = window.RMP || {};
      window.RMP.Settings = {
        pathname : "",
        uri : "/member",
        page: "account_page",
        typeahead :"http://search.mtvnservices.com/typeahead/suggest/?solrformat=true&rows=10",
        professorID : "",
        schoolID: "",
        department: "",
        state: "",
        country: "",
        user: {},
        pageLevelData: {
  "pageName": "#",
  "channel": "login",
  "hier2": "login/main",
  "prop8": "visitor"
},
        reloadInterval: 30000
      }
    </script> 
 
<script type="text/javascript">
      if (typeof MTVN == "undefined") MTVN = {};
      if (typeof MTVN.config == "undefined") MTVN.config = {};
      if (typeof MTVN.config.btg == "undefined") MTVN.config.btg = {};
      if (typeof MTVN.config.btg.DoubleClick == "undefined") MTVN.config.btg.DoubleClick = {};

      MTVN.config.btg.DoubleClick.auto = false;
      MTVN.config.btg.DoubleClick.onDemand = true;
      MTVN.config.btg.DoubleClick.reloadableAds = true;
      MTVN.config.btg.DoubleClick.reloadInterval = 30000;
      //temporary setting for testing refresh by visibility only on search page
      // if(RMP.Settings.page == 'search_results' || RMP.Settings.page == 'teacher_results')
      //   var autoRefreshByVisibility = true;
      // else
      var autoRefreshByVisibility = false;
      console.log("Refresh By Visibility: " + (autoRefreshByVisibility ? 'On' : 'Off'))

      var vmn_page_data = {
        "metadata" :  { "property" : "rmp", "hub" : "", "is_main": false },
        "mappings" : {
          "ad-unit" : {"1" : "@property", },
          "key-values" : {
  "pageName": "#",
  "channel": "login",
  "hier2": "login/main",
  "prop8": "visitor"
},
          "exclusion-categories" : "hub_pages"
        },
        "settings" : {
          "reload" : true,
          "reload_interval": 30000,
          "autoRefreshByVisibility" : autoRefreshByVisibility
        }
      };
    </script> 

<script type="text/javascript" src="assets/myjs/index1.js"></script>

<script type="text/javascript" src="assets/myjs/index2.js"></script>

<script language='JavaScript' type='text/javascript'>
   	  // This needs to be enabled in RMP's environment
      mtvn.btg.config.ReportSettings.Omniture={ enable:true, account:'viaviarmp', trackExternalLinks:true };
    </script> 
 

<script src="assets/application-4c66987976b8fccf376e7aac9add4986.js" type="text/javascript" ></script> 
<script src="assets/libs/utils-8a88ae62ab3a733b877e63de32541da6.js" type="text/javascript" ></script> 

</body>

</html>
<?php

?>
